### ISSUE TITLE
Many lines of different colors on one person

### ISSUE THREAD

**Issue by shenjunyuqwert** [TeCSAR-UNCC/EfficientHRNet#9]
Dear authon:
It is an excellent work. I am very appreciate for your selfless dedication.However  i have a question. we know one color represent one person.But why are there many lines of different colors on one plain person no matter which models of H0 H-1 H-2 H-3 or H-4 during demoing. In addition,Why are many joints predicted inversely? Is the function of tag not strong enough or the problem of post-processing? Can increasing the network for generating tag solve this problem？I would very appreciate it if you can give me a reply.

**Comment by shenjunyuqwert** [comment]
Even in your paper, there are the following situations
![image](https://user-images.githubusercontent.com/59952043/146627107-66a60052-ec25-42d0-a018-a91278e9ea22.png)

**Comment by ghost** [comment]
This issue is largely caused by the grouping algorithm, which is done in post processing (the same algorithm that was introduced in Associative Embedding, check the paper for the citation). We did nothing to change this, but we have noticed that while the keypoints are normally detected well, the grouping of them is often weak. One way to solve this is to use a different method for grouping. A recent paper called LOGO-CAP is actually doing the grouping as part of the network itself and seems to have pretty solid results on COCO. As a side note, we've seen this issue become significantly worse when inferencing on real-world data that is different from the data that it was trained on (i.e. COCO).

The change in color is because two people are detected basically on top of each other. What is actually happening is the network is detecting the same person twice and not filtering the duplicate detections out because one or more keypoints are more distant than the internal filtering threshold. In our real-world implementation, we add additional filtering to our post processing, but for fair comparisons in the paper we kept everything other than the actual network the same as in other works.

There's more that can explain this, and it largely has to do with COCO's validation methodology, which seems to have been set up more for top-down approaches than bottom-up. We're submitting a follow up paper where we try to address some of these issues and make these networks more deployable for real-world applications, but the main takeaway is that validation accuracy on datasets does not always translate to accuracy in the real-world.

I hope I answered your question? Sorry for the late response, but let me know if I can clear anything else up.

**Comment by shenjunyuqwert** [comment]
Thanks for your reply.Your network structure is amazing.

**Comment by shenjunyuqwert** [comment]
Can you clarify which paper logo-cap refers to？

**Comment by ghost** [comment]
Here's the arXiv link.

https://arxiv.org/abs/2109.03622
